core/include/clist.h

```diff
@@ -40,23 +93,19 @@
  *
  * Used as is as reference to a list.
  *
- * clist stores a pointer to the last element of a list.  That way, both
- * appending to end of list and removing head can be done in constant time.
- *
- * Actual list objects should have a @c clist_node_t as member and then use
- * the container_of() macro in list operations.
- * See @ref thread_add_to_list() as example.
  */
 typedef list_node_t clist_node_t;
 
 /**
- * @brief inserts *new_node* into *list*
+ * @brief Appends *new_node* at the end of *list
+ *
+ * @note Complexity: O(1)
```

Comment:
If you don't insist, I would keep it ascii, as some people do look at the code to read the documentation, and in that case ```O(n)``` is a lot more pleasant to the eyes than ```thcal{O}(n)$```.